config.go

```diff
@@ -80,3 +81,17 @@ func WithFilter(filter func(r *http.Request) bool) Option {
 		cfg.Filter = filter
 	})
 }
+
+// WithTraceResponseHeaderKey is used for changing response header key that contains trace id.
+// Instead of using a fixed name as the parameter, this function accepts a function
+// that takes a string (the default trace header) and returns a string (the custom trace header).
+// If the provided function is nil, the default trace header (X-Trace-ID) will be used.
+func WithTraceResponseHeaderKey(headerKeyFunc func() string) Option {
```

Comment:
I think we should rename `WithTraceResponseHeaderKey` option into `WithTraceIDResponseHeader`.

By doing this we know that this option will be used to enable adding trace id in the response header.